src/vm/methodtable.cpp

```diff
@@ -2627,15 +2644,10 @@ bool MethodTable::ClassifyEightBytesWithNativeLayout(SystemVStructRegisterPassin
 #ifdef _DEBUG
     LOG((LF_JIT, LL_EVERYTHING, "%*s**** Classify for native struct %s (%p), startOffset %d, total struct size %d\n",
         nestingLevel * 5, "", this->GetDebugClassName(), this, startOffsetOfStruct, helperPtr->structSize));
-    int fieldNum = -1;
 #endif // _DEBUG
 
-    while (numIntroducedFields--)
+    for (unsigned int fieldNum = 0; fieldNum < numIntroducedFields; fieldNum++)
```

Comment:
Can we change `fieldNum` to `fieldIndex`? When I see 'num' I think of number/count and that isn't what this means in this case. See `numIntroducedFields` as an example.